war/src/main/java/io/lumeer/engine/controller/ViewFacade.java

```diff
@@ -59,48 +57,62 @@
    @Inject
    private SecurityFacade securityFacade;
 
+   @Inject
+   private ProjectFacade projectFacade;
+
    /**
     * Creates initial metadata for the view
     *
     * @param originalViewName
     *       name given by user
     * @param viewType
     *       type of the view
+    * @param description
+    *       view description
     * @param configuration
     *       configuration of the view
     * @return view id
     * @throws ViewAlreadyExistsException
     *       when view with given name already exists
     */
-   public int createView(final String originalViewName, final String viewType, final DataDocument configuration) throws ViewAlreadyExistsException {
-      if (!dataStorage.hasCollection(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME)) {
-         dataStorage.createCollection(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME);
+   public int createView(
+         final String originalViewName,
+         final String viewType,
+         final String description,
+         final DataDocument configuration) throws ViewAlreadyExistsException {
+
+      if (!dataStorage.hasCollection(metadataCollection())) {
+         dataStorage.createCollection(metadataCollection());
          // we create indexes on frequently used fields
          int indexType = LumeerConst.Index.ASCENDING;
-         dataStorage.createIndex(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME, new DataDocument(LumeerConst.View.VIEW_ID_KEY, indexType), false);
-         dataStorage.createIndex(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME, new DataDocument(LumeerConst.View.VIEW_TYPE_KEY, indexType), false);
-         dataStorage.createIndex(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME, new DataDocument(LumeerConst.View.VIEW_NAME_KEY, indexType), false);
+         dataStorage.createIndex(metadataCollection(), new DataDocument(LumeerConst.View.ID_KEY, indexType), false);
+         dataStorage.createIndex(metadataCollection(), new DataDocument(LumeerConst.View.TYPE_KEY, indexType), false);
+         dataStorage.createIndex(metadataCollection(), new DataDocument(LumeerConst.View.NAME_KEY, indexType), false);
       }
```

Comment:
You should set this indexes as unique and remove `TYPE_KEY` index because of Cardinality.